glslc/src/file_includer.h

```diff
@@ -45,6 +49,7 @@ class FileIncluder : public shaderc::CompileOptions::IncluderInterface {
   shaderc_includer_response response_;
   std::vector<char> file_content_;
   std::string file_full_path_;
+  std::set<std::string> file_path_trace_;
```

Comment:
I think the order doesn't matter, so can we use 'unordered_set' instead? I used 'set' as it can ease a little bit on the testing side as we are using string comparison and order does matter in such case. But we can always change this.